client/components/controllers/Controller.js

```diff
@@ -87,8 +93,8 @@ class Controller extends Component {
   }
 
   componentWillUnmount() {
+    socket.removeAllListeners()
```

Comment:
DId you ever add  a console log here to see if it actually gets called?  Maybe all our assumptions are wrong...